### PR TITLE
make crossguid a real library by providing a proper SONAME

### PR DESCRIPTION
**Pull request by olafhering**
Signed-off-by: Olaf Hering <olaf@aepfle.de>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
 
 option(CROSSGUID_TESTS "Build test runner" ON)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (asmaloney) [inline CMakeLists.txt]: This should really be in a separate PR as it isn't related to this change.

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +20,12 @@ target_include_directories(crossguid PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>)
 
+set(MYLIB_VERSION_MAJOR 0)
```
Comment (asmaloney) [inline CMakeLists.txt]: Should be `CROSSGUID_VERSION_*` not `MYLIB_VERSION_*`

**Comment by olafhering** [comment]
I have updated the change to use `project()` to set the version. SOVERSION still has to be set.
